tools/dm/manually-handling-sharding-ddl-locks.md

```diff
@@ -0,0 +1,436 @@
+
+After `DM-worker-2` restarts, it will try to synchronize the waiting DDL lock before it restarted. At this point, a new lock will be created on `DM-master`, and the DM-worker will become the owner of the lock (other DM-workers have executed/skipped the DDL by now and are continuing the synchronization process).
+
+The operation processes are:
+
+1. Use `show-ddl-locks` to confirm if the corresponding lock of the DDL exists on `DM-master`. 
+    
+    Only the restarted DM-worker (`127.0.0.1:8263`) is at the `synced` state. 
+    
+    ```bash
+    » show-ddl-locks
+    {
+        "result": true,
+        "msg": "",
+        "locks": [
+            {
+                "ID": "test-`shard_db`.`shard_table`",
+                "task": "test",
+                "owner": "127.0.0.1:8263",
+                "DDLs": [
+                    "USE `shard_db`; ALTER TABLE `shard_db`.`shard_table` ADD COLUMN `c2` int(11);"
+                ],
+                "synced": [
+                    "127.0.0.1:8263"
+                ],
+                "unsynced": [
+                    "127.0.0.1:8262"
+                ]
+            }
+        ]
+    }
+    ```
+        
+2. Use `unlock-ddl-lock` to ask `DM-master` to unlock the lock.
+    
+    - Use the parameter `--worker` to limit the operation to only target at the restarted DM-worker (`127.0.0.1:8263`). 
+    - The DM-worker will try to execute the DDL to the downstream again during the unlocking process (the owner before restarting has executed the DDL to the downstream), so as to make sure that the DDL can be executed multiple times.
+    
+        ```bash
+        » unlock-ddl-lock --worker=127.0.0.1:8263 test-`shard_db`.`shard_table`
+        {
+            "result": true,
+            "msg": "",
+            "workers": [
+                {
+                    "result": true,
+                    "worker": "127.0.0.1:8263",
+                    "msg": ""
+                }
+            ]
+        }
+        ```
+        
+3. Use `show-ddl-locks` to confirm if the DDL lock has been successfully unlocked.
+4. Use `query-status` to confirm if the synchronization task is normal.
+
+#### Impact 
+
+After manually unlocking the lock, the following sharding DDL can be synchronized automatically and normally.
+
+### Scenario 3: Some DM-workers are temporarily unreachable during the DDL unlocking process
+
+#### The reason for the abnormal lock
+
+It has the similar reason for the abnormal lock in [Some DM-workers restart during the DDL unlocking process](#scenario-2-some-dm-workers-restart-during-the-ddl-unlocking-process). If the DM-worker is temporarily unreachable when you ask the DM-worker to skip the DDL, this DM-worker might fail to skip the DDL.
```

Comment:
same with `Scenario 3`,  we can also use solution of `Scenario 3` to solve problem of `Scenario 2`

Reply:
correct it, the difference is dm-master doesn't have a lock in  Scenario 3, but dm-master has a new lock lock in Scenario 2